client-tags/typing.md

```diff
@@ -0,0 +1,51 @@
+---
+title: IRCv3 typing client tag
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "evan@muffinmedic.net"
+---
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+typing` tag name. Instead, implementations SHOULD use the
+`+draft/typing` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Description
+This specification defines the `typing` client tag, which allows clients to send and receive the current status of when a user is typing a new message prior to sending.
+
+For the purpose of this specification, a message is defined as a user-generated `PRIVMSG`, `NOTICE`, `TAGMSG`, or [`batch`][batch] end.
+
+## Format
+The `typing` tag is sent as a [`TAGMSG`][tags]. The client tag takes no parameters.
+
+The `typing` client tag SHOULD be sent by the client when the first character is typed in the text-input field. If new text is entered into the text-input field following sending of the text-input field content, the client MUST send a new `typing` tag after sending the preceding content.
```

Comment:
The client needs to keep sending typing notifications every (say) 3 seconds while typing, otherwise the receiving end will always time it out 10 seconds after the sender starts typing.